src/SdfEntityCreator.cc

```diff
@@ -481,10 +531,35 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Joint *_joint)
       components::Name(_joint->Name()));
   this->dataPtr->ecm->CreateComponent(jointEntity ,
       components::ThreadPitch(_joint->ThreadPitch()));
-  this->dataPtr->ecm->CreateComponent(jointEntity,
-      components::ParentLinkName(_joint->ParentLinkName()));
-  this->dataPtr->ecm->CreateComponent(jointEntity,
-      components::ChildLinkName(_joint->ChildLinkName()));
+
+  std::string resolvedParentLinkName;
+  const auto resolveParentErrors =
+    _joint->ResolveParentLink(resolvedParentLinkName);
+  if (!resolveParentErrors.empty())
+  {
+    ignerr << "Failed to resolve parent link for joint '" << _joint->Name()
+           << "' with parent name '" << _joint->ParentLinkName() << "'"
+           << std::endl;
+
+    return kNullEntity;
+  }
+  this->dataPtr->ecm->CreateComponent(
+      jointEntity, components::ParentLinkName(resolvedParentLinkName));
+
+  std::string resolvedChildLinkName;
+  const auto resolveChildErrors =
+    _joint->ResolveChildLink(resolvedChildLinkName);
+  if (!resolveChildErrors.empty())
+  {
+    ignerr << "Failed to resolve child link for joint '" << _joint->Name()
+           << "' with child name '" << _joint->ChildLinkName() << "'"
+           << std::endl;
```

Comment:
Consider printing `resolveChildErrors` as well since it might contain additional details about the error.